### ISSUE TITLE
Add combinational logic keyword rules

### ISSUE THREAD

**Issue by jscottee** [camsteffen/polite#10]
The current version (1.4.7, updated on Oct. 16, 2019) allows for keyword matching and inverse keyword matching, but not both. It would be nice to establish a hierarchy of that activates a certain set of keywords _except_ for when another set of keywords are present.

**Comment by camsteffen** [comment]
How would you use this feature? It's a neat idea but I wonder if it is practically useful.

**Comment by jscottee** [comment]
I may have bitten off more than I could chew here, lol. I can think of a very specific use case that applies only to me, but it might be overkill for the majority of users.

For the sake of argument, I often have the word "call" in multiple contexts—either to silence my phone for a professional call that I'm going to receive, or as a reminder to call someone (professional or otherwise). It would be nice to put "reminder" in the title and have Polite ignore that event on my calendar. (I'm a noob and use G Cal events as reminders.)

Generally, we would use it as "trigger when ABC keyword is present, but ignore if XYZ word is also present."

**Comment by camsteffen** [comment]
> I may have bitten off more than I could chew here, lol.

😆 That's okay. I can just leave the issue open for future discussion.

> Generally, we would use it as "trigger when ABC keyword is present, but ignore if XYZ word is also present."

So, in your example, what is ABC and what is XYZ?

**Comment by jscottee** [comment]
> So, in your example, what is ABC and what is XYZ?

I'll rephrase it—trigger when "call" is a keyword, but, say, long press on that keyword to add exceptions, like "social". So, add exceptions to individual keywords. That seems like a reasonable iteration.

**Comment by camsteffen** [comment]
So you would have calendar events with the word "social"? I'm trying to understand this as a concrete use case.

**Comment by jscottee** [comment]
Correct. In addition, say I have a regularly occurring call with my family (I do) that I don't want to trigger Polite. My calendar event title is "Zoom call w/ fam". Because "fam" is idiosyncratic and does not often appear in other , I would want to alter the "call" keyword trigger to exclude events with the word "fam" in it.

To reiterate, the Polite app is excellent, and there are only a handful of events in which I regularly dismiss Polite as soon as it starts. Adding exceptions would just account for more of these edge cases.

**Comment by skaldebane** [comment]
Just sayin', but I'm wondering if this can be generalized using Regex... Is that a possibility? That would absolutely make it very powerful and useful for every use case.
